torch/csrc/api/include/torch/data/ctf/ctf_parser.h

```diff
@@ -0,0 +1,931 @@
+#pragma once
+
+#include <torch/data/ctf/reader.h>
+#include <torch/data/ctf/utils.h>
+
+#include <stdint.h>
+#include <algorithm>
+#include <cassert>
+#include <climits>
+#include <iostream>
+#ifdef CTF_DEBUG
+#include <map>
+#endif
+#include <memory>
+#include <ostream>
+#include <unordered_map>
+#include <vector>
+
+namespace torch {
+namespace data {
+namespace ctf {
+
+/*
+ * CTF general format
+ * [Sequence_Id](Sample or Comment)+
+ *   where
+ *          sequence_Id=(empty|[0-9]+)
+ *          Sample=|Input_Name (Value )*
+ *          Comment=|# some content
+ * Example:
+ * 100 |a 1 2 3 |b 100 200
+ * 100 |a 4 5 6 |b 101 201
+ * 100 |b 102983 14532 |a 7 8 9
+ * 100 |a 7 8 9
+ * 200 |b 300 400 |a 10 20 30
+ * 333 |b 500 100
+ * 333 |b 600 -900
+ * 400 |a 1 2 3 |b 100 200
+ * |a 4 5 6 |b 101 201
+ * |a 4 5 6 |b 101 201
+ * 500 |a 1 2 3 |b 100 200
+ */
```

Comment:
I like this example. Very helpful on understanding this format. Maybe we can also add a line with comment as well.

Reply:
I will add some comments and a sparse input stream to demonstrate how generic it can be. Also some empty input streams